pkg/controllers/sandbox/restartsandbox_reconciler.go

```diff
@@ -0,0 +1,114 @@
+/*
+ (c) Copyright [2021-2024] Open Text.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package sandbox
+
+import (
+	"context"
+	"errors"
+
+	"github.com/go-logr/logr"
+	v1 "github.com/vertica/vertica-kubernetes/api/v1"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	vdbcontroller "github.com/vertica/vertica-kubernetes/pkg/controllers/vdb"
+	verrors "github.com/vertica/vertica-kubernetes/pkg/errors"
+	"github.com/vertica/vertica-kubernetes/pkg/vdbstatus"
+	"github.com/vertica/vertica-kubernetes/pkg/vk8s"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// RestartSandboxReconciler will make sure the sandbox and its subclusters'
+// shutdown states are properly set after sandbox restart.
+type RestartSandboxReconciler struct {
+	SRec      *SandboxConfigMapReconciler
+	Vdb       *v1.VerticaDB
+	Log       logr.Logger
+	PFacts    *vdbcontroller.PodFacts
+	RestartSC bool
+}
+
+func MakeRestartSandboxReconciler(r *SandboxConfigMapReconciler,
+	vdb *v1.VerticaDB, pfacts *vdbcontroller.PodFacts, log logr.Logger) controllers.ReconcileActor {
+	return &RestartSandboxReconciler{
+		SRec:   r,
+		Log:    log.WithName("RestartSandboxReconciler"),
+		Vdb:    vdb,
+		PFacts: pfacts,
+	}
+}
+
+func (r *RestartSandboxReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
+	if res, err := r.updateSandboxShutdownState(ctx); verrors.IsReconcileAborted(res, err) {
```

Comment:
Add r.PFacts.Collect() here since the previous reconciler could invalidate the pod facts.

Reply:
I collect pfacts at line 76.